scripts/format_arpa_lm.py

```diff
@@ -41,13 +42,45 @@
     sys.exit("format_arpa_lm.py: expected directory {0} to exist.".format(
             args.temp_dir))
 
+# verify the input string max_memory
+if args.max_memory != '':
+    # valid string max_memory must have at least two items 
+    if len(args.max_memory) >= 2:
+        s = args.max_memory
+        # case1: valid string max_memory can be formatted as "integer + letter or '%'"
+        # the unit of memory size can also be 'T', 'P', 'E', 'Z', or 'Y'. They
+        # are not included here considering their rare use in practice
+        if s[-1] in ['b', '%', 'K', 'M', 'G']:
+            if not s[-2].isdigit():
```

Comment:
you're right. fixed it.

On Sat, Aug 13, 2016 at 12:05 PM, Wang Jian <notifications@github.com>
wrote:

> In scripts/format_arpa_lm.py
> <https://github.com/danpovey/pocolm/pull/46#discussion_r74688663>:
>
> > @@ -41,13 +42,45 @@
> >      sys.exit("format_arpa_lm.py: expected directory {0} to exist.".format(
> >              args.temp_dir))
> >
> > +# verify the input string max_memory
> > +if args.max_memory != '':
> > +    # valid string max_memory must have at least two items
> > +    if len(args.max_memory) >= 2:
> > +        s = args.max_memory
> > +        # case1: valid string max_memory can be formatted as "integer + letter or '%'"
> > +        # the unit of memory size can also be 'T', 'P', 'E', 'Z', or 'Y'. They
> > +        # are not included here considering their rare use in practice
> > +        if s[-1] in ['b', '%', 'K', 'M', 'G']:
> > +            if not s[-2].isdigit():
>
> Why only check s[-2].isdigit()? I think all char in s[:-1] should be a
> digit.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/danpovey/pocolm/pull/46/files/b0337e27cb3436df20b6efbf1ced644a7e46e566..d45eba31cad33e1e0b101c6a94beb73fc5acf4cd#r74688663>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ANVxSiGBaF0jFQHYCy65fLYlXpd3aEspks5qfetBgaJpZM4JibxT>
> .
>



-- 
Ke Li
Dept. of Electrical and Computer Engineering
Johns Hopkins University
Email: kli26@jhu.edu